lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1252,19 +1255,16 @@ private synchronized void onConnected(ProtocolMessage message) {
     private void addPendingMessagesToQueuedMessages(boolean resetMessageSerial) {
         // Add messages from pending messages to front of queuedMessages in order to retry them
         queuedMessages.addAll(0, pendingMessages.queue);
-        //rewind start serial back to the first serial since we are clearing the queue
-        if (!pendingMessages.queue.isEmpty()){
-            //Reset current serial to the first pending message on previous queue as we are going to clear the queue now
+        //rewind start serial back to the first serial since we are clearing the queue if the queue is not empty
```

Comment:
I think this would be a lot clearer if it was:

```
if (resetMessageSerial){  // failed resume, so all new published messages start with msgSerial = 0
  msgSerial = 0; //msgSerial will increase in sendImpl when messages are sent
  pendingMessages.resetStartSerial(0);
} else if(!pendingMessages.queue.isEmpty()) { // pendingMessages needs to expect next msgSerial to be the earliest previously unacknowledged message
  msgSerial =  pendingMessages.queue.get(0).msg.msgSerial;
  pendingMessages.resetStartSerial((int) (msgSerial));
}

```

Reply:
57c9a142260dd198c40f054787a7aea6cb0d9542